napari/plugins/io.py

```diff
@@ -280,12 +282,106 @@ def _is_null_layer_sentinel(layer_data: Union[LayerData, Any]) -> bool:
     )
 
 
+def _write_layers_with_npe2(
+    path: str, layers: List[Layer], plugin_command_name: Optional[str] = None
+) -> List[Optional[str]]:
+    """
+    Write layers to a file using an NPE2 plugin.
+
+    Parameters
+    ----------
+    path : str
+        The path (file, directory, url) to write.
+    layer_type : str
+        All lower-class name of the layer class to be written.
+    plugin_name : str, optional
+        Name of the plugin to write data with. If None then all plugins
+        corresponding to appropriate hook specification will be looped
+        through to find the first one that can write the data.
+
+
+    Returns
+    -------
+    None or list of str
+        None when no plugin was found, otherwise a list of file paths, if any,
+        that were written.
+    """
+
+    logger.debug("Saving with NPE2")
+
+    try:
+        import npe2
+        from npe2.manifest.io import WriterContribution
+    except ImportError:
+        return [None]
+
+    layer_data = [layer.as_layer_data_tuple() for layer in layers]
+    layer_types = [ld[2] for ld in layer_data]
+
+    def _lookup_writer(
+        path: str,
+    ) -> Tuple[Optional[WriterContribution], str, str]:
+        """Returns (writer,new_path,ext)
+
+        When the path has a file extension, find a compatible writer that has
+        that same extension.
+
+        When there is no extension and only a single layer, find a compatible
+        writer and append the extension.
+
+        Otherwise, find a compatible no-extension writer and write to that.
+        No-extension writers typically write to a folder.
+        """
+        ext = os.path.splitext(path)[1].lower() if path else ''  # type: str
```

Comment:
lets prefer type hints to type comments:
```suggestion
ext: str = os.path.splitext(path)[1].lower() if path else ''
```

(but also ... fwiw, this is an unnecessary one since both `lower()` and `''`are str)